File: go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	// panic("Implement the Value function")
	resistorValue := 0

	for index, value := range colors {
		if index == 1 {
			resistorValue += resistorColor[value]
			break
		} else if index == 2 {
			break
		}
		resistorValue += resistorColor[value] * 10
	}

	return resistorValue
}

var resistorColor = map[string]int{
	"black":  0,
	"brown":  1,
	"red":    2,
	"orange": 3,
	"yellow": 4,
	"green":  5,
	"blue":   6,
	"violet": 7,
	"grey":   8,
	"white":  9,
}

File: typescript/resistor-color-duo/resistor-color-duo.ts
export function decodedValue(colors: string[]): number {
  let index = -1
  let counter = 0
  let value = 0

  for (let color of colors) {
    index = COLORS.findIndex(c => c === color.toLowerCase())
    counter++;
    if (index === -1){
      throw new Error('invalid color: ' + color)
    }
    if (counter ===2) {
      value += index
    }else if (counter === 1) {
      value = index*10
    } 
  }

  return value
}


export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
] as const;
